### PR TITLE
Don't run CI for docs-only changes

### LINKED ISSUE #58710: A possible easy solution to not build docs-only changes on CI
This doesn't change yaml files, only changes on dev.azure.com.

image

I'm not sure if this will work as intended, but I tested on a small repo, and it seems to work?

image

The check is required and it's not showing the "Use your administrator privileges to merge this pull request without waiting for requirements to be met." message.

@jaredpar Would you be able to test this and see if it works?

### PR DESCRIPTION
**Pull request by Youssef1313**
Closes #58710
Per https://github.com/dotnet/roslyn/pull/58716#issuecomment-1007736038

### CODE REVIEW

File: azure-pipelines-integration-corehost.yml [hunk 1]
```diff
@@ -12,11 +12,17 @@ trigger:
 # Automatic building of all PRs is disabled in the pipeline's trigger page.
 # See https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#comment-triggers
 pr:
-- main
-- main-vs-deps
-- release/*
-- features/*
-- demos/*
+  branches:
```
Comment (RikkiGibson) [inline azure-pipelines-integration-corehost.yml:15]: I don't see a need to try it immediately but: there's no way to template this section out, correct?

**Comment by RikkiGibson** [comment]
Just want to see https://github.com/dotnet/roslyn/pull/58716#partial-pull-merging make more progress out of paranoia, then will sign off.

File: azure-pipelines-integration-corehost.yml (hunk 1 above)
Comment (Youssef1313) [reply azure-pipelines-integration-corehost.yml:15]: No idea if that's possible tbh.

**Review by RikkiGibson — APPROVED** [review]
Thanks @Youssef1313!

**Pull request merged**